pkg/infradb/taskmanager/taskmanager_test.go

```diff
@@ -0,0 +1,295 @@
+
+		task := TaskMan.taskQueue.Dequeue()
+
+		assert.NotNil(t, task, "Task should not be nil")
+		assert.Equal(t, "testTaskEventError", task.name, "Task name should match")
+		assert.Equal(t, "testTypeEventError", task.objectType, "Task object type should match")
+		assert.Equal(t, "testVersionEventError", task.resourceVersion, "Task resource version should match")
+		assert.Equal(t, subscribers, task.subs, "Task subscribers should match")
+	}
+
+	func TestTaskManager_Timeout(t *testing.T) {
+		var wg sync.WaitGroup
+
+		handler := &ModulelciHandler{}
+		TaskMan.StartTaskManager()
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			eventbus.EBus.StartSubscriber("testModuleTimeout", "testEventTimeout", 1, handler)
+		}()
+
+		// Wait for both the TaskManager and the subscriber to be started
+		wg.Wait()
+		subscribers := eventbus.EBus.GetSubscribers("testEventTimeout")
+		if len(subscribers) == 0 {
+			t.Fatalf("No subscribers found for event type 'testEventTimeout'")
+		}
+		TaskMan.CreateTask("testTaskTimeout", "testTypeTimeout", "testVersionTimeout", subscribers)
+
+		time.Sleep(35 * time.Second)
+
+		task := TaskMan.taskQueue.Dequeue()
+
+		assert.NotNil(t, task)
+		assert.Equal(t, "testTaskTimeout", task.name)
+		assert.Equal(t, "testTypeTimeout", task.objectType)
+		assert.Equal(t, "testVersionTimeout", task.resourceVersion)
+		assert.Equal(t, subscribers, task.subs)
+	}
+
+	func TestTaskManager_NotificationIdNotMatching(t *testing.T) {
+		var wg sync.WaitGroup
+
+		// Start the TaskManager in a separate goroutine
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			TaskMan.StartTaskManager()
+		}()
+
+		handler := &ModulelciHandler{}
+		// Start the subscriber and wait for it to be ready
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			eventbus.EBus.StartSubscriber("testModuleNotificationIdNotMatching", "testEventNotificationIdNotMatching", 1, handler)
+
+		}()
+
+		// Wait for both the TaskManager and the subscriber to be started
+		wg.Wait()
+
+		for i := 0; i < cap(TaskMan.taskStatusChan); i++ {
+			TaskMan.taskStatusChan <- &TaskStatus{}
+		}
+
+		subscribers := eventbus.EBus.GetSubscribers("testEventNotificationIdNotMatching")
+		if len(subscribers) == 0 {
+			t.Fatalf("No subscribers found for event type 'testEvent'")
+		}
+		TaskMan.CreateTask("testTaskNotificationIdNotMatching", "testTypeNotificationIdNotMatching", "testVersionNotificationIdNotMatching", subscribers)
+
+		time.Sleep(500 * time.Millisecond)
+
+		select {
+		case task := <-TaskMan.taskQueue.channel:
+			if task.name == "testTask" {
+				t.Errorf("assert failed:")
+			}
+		default:
+		}
+
+		for len(TaskMan.taskStatusChan) > 0 {
+			<-TaskMan.taskStatusChan
+		}
+	}
+*/
+func TestTaskManager_StatusUpdatedChannelNotAvailable(t *testing.T) {
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+
+	go handleTestEventbusy(&wg)
+
+	wg.Wait()
+
+	retValMu.Lock()
+	actualRetVal := retVal
```

Comment:
atomic?

Reply:
i believe we can keep lock as its limited to this test only.

Reply:
You use mutex and workgroup to find out if your var was changed. I think it is overkill. Using a chan would be enough (you pass a chan into `handleTestEventbusy`, you wait in select for chan be not empty or timeout, `handleTestEventbusy` sends into chan when ready)